src/main/java/bridge/util/constant/MoveKey.java

```diff
@@ -0,0 +1,31 @@
+package bridge.util.constant;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum MoveKey {
+    DOWN("D", 0),
+    UP("U", 1);
+
+    private final String moveKey;
+    private final int line;
+
+    MoveKey(String moveKey, int line) {
+        this.moveKey = moveKey;
+        this.line = line;
+    }
+
+    public String getMoveKey() {
+        return moveKey;
+    }
+
+    public int getLine() {
+        return line;
+    }
+
+    public static Optional<MoveKey> getEnum(String moveKey) {
+        return Arrays.stream(values())
+                .filter(key -> key.getMoveKey().equals(moveKey))
+                .findFirst();
+    }
```

Comment:
여기 갑자기 생각난건데 Optional로 해서 findFirst() 없으면 Null 이 저장되는거 아닌가요 ? 나중에 orElseThrow로 처리 하는데 그렇게 처리하면 다른 키 입력에 대해선 예외처리가 제대로 진행되나요 ? 굳이 Optional 이용해서 null 허용해준 이유가 궁금합니다
```suggestion
    public static MoveKey getEnum(String moveKey) {
        return Arrays.stream(values())
                .filter(key -> key.getMoveKey().equals(moveKey))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException();
    }
```
이렇게 사용해도 되지않나요 ?

저는 이런 방식으로 getEnum에서 예외처리까지 맡아주는게 좋다고 생각합니다 !

Reply:
민혁님이 제시해주신 코드가 좀 더 알맞게 책임 분리가 될거 같다는 생각이 들었어요!
이 부분도 한번 수정을 해보겠습니다

Reply:
`while(moveKey == null)` 부분은 음.. 개인의 선택인거 같긴한데 제 생각입니다만 while문을 사용하는 이유는 예외가 발생했을 경우에 반복문을 통해서 다시 실행시키기 위함으로 보입니다.
그런데 try-catch문을 통해서 catch문에서 try 문에서 예외가 발생하면 실행시킬 행동을 정의해준다고 생각하는데 여기서 재귀호출을 사용한다면 굳이 while문을 사용할 필요가 있을까요 ? 

개인적인 생각입니다만 try-catch만으로도 할 수 있는 행동을 이미 try-catch를 사용하고 있음에도 불구하고 굳이 while문으로 감싸서 작성한느낌(?) 이라서 재귀로 하는게 어떨까 라고 얘기드린거였습니다 !
그리고 `while(true)` 보단 확실히 조건문에 어떤 조건이 들어가있는게 좋다고 생각하긴합니다 ! 준호님이 말씀해주신 `while(moveKey == null)` 처럼 조건이 있으면 좀 더 읽기 좋은 것 같아요

Reply:
민혁님이 남겨주신 글을 보고 아이디어가 하나 떠올랐습니다..! 한번 trr-catch없이 while로만 리팩토링을 해볼게요..!